daffodil-core/src/main/scala/org/apache/daffodil/dsom/SimpleTypes.scala

```diff
@@ -221,64 +311,399 @@ final class LocalSimpleTypeDef(
+              case Some(r) => r.diagnosticDebugName
+              case None    => repType.toString()
+            }
+            context.SDE(
+              "repType (%s) with primitive type (%s) used without defining a transformation is not compatable with the baseType of (%s) with primitive type (%s)",
+              repTypeName, srcType.name,
+              diagnosticDebugName, dstType.name)
+          }
+          None
+        }
+        case (Some(_), Some(_), _) => Assert.invariantFailed("Cannot combine an enumeration with a union")
+        case (Some(_), _, Some(_)) => context.SDE("Cannot use typeCalcExpressions while defining repValues of enumerations")
+        case (_, Some(_), Some(_)) => context.SDE("Cannot use typeCalcExpressions while using a union that defines typeCalcs")
+      }
+
+      ans match {
+        case Some(idt: IdentifyTypeCalculator[AnyRef]) => {
+          if (srcType != dstType) {
+            SDE("Identity transform requires that the basetype and reptype have a common primitive type")
+          }
+        }
+        case _ => ()
+      }
+
+      ans
+
+    })
+  }
+
+  /*
+   * We don't really need the NamedMixin. It is only used for detecting duplicates
+   * However, since only named types can be a repType, there is no problem
+   * in requiring them to be named
+   */
+  override lazy val optRepTypeFactory: Option[SimpleTypeFactory with NamedMixin] = {
+    lazy val fromSelf: Option[SimpleTypeFactory with NamedMixin] = {
+      val qName = findPropertyOptionThisComponentOnly("repType").toOption
+        .flatMap(QName.resolveRef(_, namespaces, tunable).toOption)
+      val optRepTypeDef = qName.flatMap(schemaSet.getGlobalSimpleTypeDef(_))
+      val optRepPrimType = qName.flatMap(schemaSet.getPrimitiveTypeFactory(_))
+      Assert.invariant(!(optRepPrimType.isDefined && optRepTypeDef.isDefined))
+      if (qName.isDefined) {
+        schemaDefinitionUnless(optRepTypeDef.isDefined || optRepPrimType.isDefined, s"Cannot find reptype ${qName.get}")
+      }
+      optRepTypeDef.orElse(optRepPrimType)
+    }
+    lazy val fromUnion: Option[SimpleTypeFactory with NamedMixin] = optUnion.flatMap(union => {
+      val repTypes = union.unionMemberTypes.map(_.optRepTypeFactory)
+      //check that all repTypes are the same
+      //Because of how we inline types, we do not expect to see structual equality,
+      //so we rely on the xml qname instead
+      val numRepTypes = repTypes.map(_.map(_.namedQName)).toSet.size
+      if (numRepTypes > 1) {
+        SDE("If any child type of a union has a repType, they all must have the same repType")
+      }
+      if (numRepTypes == 0) {
+        None
+      } else {
+        repTypes.head
+      }
+    })
+    fromSelf.orElse(fromUnion)
+  }
+
+  override lazy val optRepValueSet: Option[RepValueSet[AnyRef]] = optRepTypeDefFactory.flatMap(repType => {
+    val primType: PrimType = repType.primType
+
+    //TODO check for SDEs
```

Comment:
Deal with these TODO's

Reply:
As far as I can tell, these TODOs have already been addressed, and I just forgot to remove the comments. Removing in next commit.